packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -709,259 +818,180 @@ public void resumeVideoRecording(@NonNull final Result result) {
-      return;
-    }
-
-    // If switching directly from torch to auto or on, make sure we turn off the torch.
-    if (flashMode == FlashMode.torch && mode != FlashMode.torch && mode != FlashMode.off) {
-      updateFlash(FlashMode.off);
+  /**
+   * Dart handler when it's time to set a new flash mode. This will try to set a new flash mode to
+   * the current camera.
+   *
+   * @param result Flutter result.
+   * @param newMode new mode.
+   */
+  public void setFlashMode(@NonNull final Result result, FlashMode newMode) {
+    // Save the new flash mode setting
+    final FlashFeature flashFeature = cameraFeatures.getFlash();
+    flashFeature.setValue(newMode);
+    flashFeature.updateBuilder(previewRequestBuilder);
 
-      this.cameraCaptureSession.setRepeatingRequest(
-          captureRequestBuilder.build(),
-          new CaptureCallback() {
-            private boolean isFinished = false;
+    refreshPreviewCaptureSession(
+        () -> result.success(null),
+        (code, message) -> result.error("setFlashModeFailed", "Could not set flash mode.", null));
+  }
 
-            @Override
-            public void onCaptureCompleted(
-                @NonNull CameraCaptureSession session,
-                @NonNull CaptureRequest request,
-                @NonNull TotalCaptureResult captureResult) {
-              if (isFinished) {
-                return;
-              }
-
-              updateFlash(mode);
-              refreshPreviewCaptureSession(
-                  () -> {
-                    result.success(null);
-                    isFinished = true;
-                  },
-                  (code, message) ->
-                      result.error("setFlashModeFailed", "Could not set flash mode.", null));
-            }
+  /**
+   * Dart handler for setting new exposure mode setting.
+   *
+   * @param result Flutter result.
+   * @param newMode new mode.
+   */
+  public void setExposureMode(@NonNull final Result result, ExposureMode newMode) {
+    final ExposureLockFeature exposureLockFeature = cameraFeatures.getExposureLock();
+    exposureLockFeature.setValue(newMode);
+    exposureLockFeature.updateBuilder(previewRequestBuilder);
 
-            @Override
-            public void onCaptureFailed(
-                @NonNull CameraCaptureSession session,
-                @NonNull CaptureRequest request,
-                @NonNull CaptureFailure failure) {
-              if (isFinished) {
-                return;
-              }
+    refreshPreviewCaptureSession(
+        () -> result.success(null),
+        (code, message) ->
+            result.error("setExposureModeFailed", "Could not set exposure mode.", null));
+  }
 
-              result.error("setFlashModeFailed", "Could not set flash mode.", null);
-              isFinished = true;
-            }
-          },
-          null);
-    } else {
-      updateFlash(mode);
+  /**
+   * Set new exposure point from dart.
```

Comment:
Sets

Please sweep method comments for the verb mood.

Reply:
Camera.java swept, other affected files still to be checked.